crates/astria-core/src/protocol/transaction/v1/action/mod.rs

```diff
@@ -1047,8 +1057,20 @@ impl Protobuf for Ics20Withdrawal {
             .transpose()
             .map_err(Ics20WithdrawalError::invalid_bridge_address)?;
 
+        if bridge_address.is_some() {
+            let parsed_withdrawal: Ics20WithdrawalFromRollup =
+                serde_json::from_str(&memo).map_err(Ics20WithdrawalError::parse_memo)?;
```

Comment:
Should we edit our domain type `Ics20Withdrawal` to then include the parsed withdrawal as well? Then, we don't have to parse again during `check_and_execute`

Reply:
Yes, absolutely. Easiest is probably to just make `Ics20Withdrawal` an enum over 2 variants (one where `bridge_address` is set and contains the parsed payload, and one where it's not set and just contains the passed through `String` memo).

You should also lean on a `serde(try_from = "")` to a) first read into some `UnverifiedIcs20WithdrawalFromRollup`, and then b) via a `TryFrom<UnverifiedIcs20WithdrawalFromRollup> Ics20WithdrawalFromRollup` come up with the final type.

Reply:
Split into enum 84b00ef. I wanted to ask further before continuing with your second suggestion: in order to do this we would need to implement a new `UnverifiedIcs20WithdrawalFromRollup` type and then the `TryFrom` trait as well for it. While this may be fine, it would be a little bit backwards from the way we currently do things. 

`Ics20WithdrawalFromRollup` is a raw protobuf type, which usually represents the "unverified" version as opposed to the verified one. We could rename it to reflect that it is unverified, and then create a new domain type for the verified version, but I'm not sure if the name change would be proto breaking. In addition to this, it seems like it would be many more lines for questionable benefit to just performing these verifications in `Ics20Withdrawal::try_from_raw`. Do you still think it would be worth it to do this?